segmentstore/server/src/main/java/io/pravega/service/server/containers/StreamSegmentContainerMetadata.java

```diff
@@ -302,7 +302,8 @@ public void setContainerEpoch(long value) {
                     .stream()
                     .filter(m -> isEligibleForEviction(m, adjustedCutoff) && !activeTransactions.contains(m.getId()))
                     .forEach(m -> {
-                        this.metadataById.remove(m.getId());
+                        StreamSegmentMetadata removedMetadata = this.metadataById.remove(m.getId());
+                        removedMetadata.markInactive();
```

Comment:
The `StreamSegmentReadIndex` and `SegmentAggregator` classes keep pointers to the objects themselves. They do not re-request them every time they need them (since they need them a lot, I did not want to slow things down with hash table lookups). As such, I need a way to notify those components that their metadata instance has become inactive and they need to do something about it (that something is close & recreate themselves, which is a mechanism that was already in place)